design/service-library.md

```diff
@@ -0,0 +1,187 @@
+# End-to-end usage of kiota-generated API Clients in JavaScript
+
+While working with the Kiota + JavaScript teams, it became clear that we needed more context and provide a better vision on how we should be using a kiota-generated API Client. This design document aims at providing clarity and a better understanding on how developers will be using our API Client and the underlying kiota packages.
+
+## Status
+
+| Date                | Version | Author           | Status |
+| ------------------- | ------- | ---------------- | ------ |
+| December 14th, 2022 | v0.3    | Sébastien Levert | Draft  | 
+| February 15th, 2022 | v0.2    | Sébastien Levert | Draft  |
+| January 17th, 2022  | v0.1    | Sébastien Levert | Draft  |
+
+## Constraints
+
+Before we jump into the end-to-end walk-through, it's important to set some constraints.
+
+| Type      | Environment | Description                                                                                       |
+| --------- | ----------- | ------------------------------------------------------------------------------------------------- |
+| Platforms | Node        | Current and 2 Previous LTS (14 / 16 / 18)                                                         |
```

Comment:
14 will be EOL in April FYI https://endoflife.software/programming-languages/server-side-scripting/nodejs

Reply:
Agree. I'll remove 14.

```suggestion
| Platforms | Node        | Current and 2 Previous LTS (16 / 18)                                                         |
```